### PR TITLE
PUBDEV-5677 - XGBoost performance logging (re-wright)

### PR DESCRIPTION
**Pull request by Pscheidl**
https://0xdata.atlassian.net/browse/PUBDEV-5677

PR to master: https://github.com/h2oai/h2o-3/pull/2611

### CODE REVIEW

**Comment by Pscheidl** [comment]
Dense logging example: 
[dense_logging_example.txt](https://github.com/h2oai/h2o-3/files/2173444/dense_logging_example.txt)

Sparse logging example:
[sparse_logging_example.txt](https://github.com/h2oai/h2o-3/files/2173445/sparse_logging_example.txt)

File: h2o-extensions/xgboost/src/main/java/ml/dmlc/xgboost4j/java/XGBoostScoreTask.java [hunk 1]
```diff
@@ -158,6 +163,12 @@ public void map(Chunk[] cs, NewChunk[] ncs) {
             }
             final float[][] preds = booster.predict(data);
 
+            stopWatch.stop();
```
Comment (michalkurka) [inline h2o-extensions/xgboost/src/main/java/ml/dmlc/xgboost4j/java/XGBoostScoreTask.java]: Logging in each map seems excessive.

File: h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoostUtils.java [hunk 2]
```diff
@@ -113,17 +114,33 @@ public static DMatrix convertFrameToDMatrix(Key<DataInfo> dataInfoKey,
...
             trainMat = new DMatrix(data, actualRows, cols, Float.NaN);
+
+            denseMatrixCopyingStopWatch.stop();
+            Log.info("Copying dense matrix with " + actualRows + " logical rows and "
+                    + cols + " logical columns "
+                    + "into native XGBoost finished in " + denseMatrixCopyingStopWatch.getTime() + " ms");
             assert trainMat.rowNum() == actualRows;
         }
 
+        StopWatch labelAndWeightsStopWatch = new StopWatch();
```
Comment (michalkurka) [inline h2o-extensions/xgboost/src/main/java/hex/tree/xgboost/XGBoostUtils.java]: Rest of the code base uses `water.util.Timer` class, I would stick to that unless there is a good reason.

**Comment by Pscheidl** [comment]
Closing this PR, this change will not be pushed into rel-wright. Targeting master.

**Pull request closed without merging**